slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/PlainSQLTest.scala

```diff
@@ -124,4 +124,35 @@ class PlainSQLTest extends AsyncTest[JdbcTestDB] {
       userForIdAndName(2, "foo").map(_.headOption shouldBe None) //TODO Support `head` and `headOption` in Plain SQL Actions
     )
   }
+
+  def testEdgeTypes = ifCap(tcap.plainSql) {
+    object Const
+    case class Weird(u: Unit, x: (Int, String), c: Const.type)
+
+    implicit val getC = GetResult.const[Const.type](Const)
+    implicit val get = GetResult[Weird](r => new Weird(r.<<[Unit], r.<<, r.<<[Const.type]))
+
+    implicit val setC = SetParameter.const[Const.type]
+    implicit val setU = SetParameter[User]{(u, p) => println(u); p.setInt(u.id); p.setString(u.name)}
+    implicit val setT = setU.contramap((User.apply _).tupled)
+    implicit val set  = SetParameter[Weird]{(w, p) =>
+      SetParameter.SetUnit(w.u, p)
+      setT(w.x, p)
+      setC(w.c, p)
+    }
+
+    val w = Weird((), (3, "Ah"), Const)
+
+    val create =
+      sqlu"create table USERS2(ID int not null primary key, NAME varchar(255))".map(_ shouldBe 0)
+
+    def testSet =
+      sqlu"insert into USERS2 values (${w.x._1}, ${w.x._2})".map(_ shouldBe 1)
+      //sqlu"insert into USERS2 values ($w)".map(_ shouldBe 1)
```

Comment:
Used to be I could write `StaticQuery[Weird, Unit]("insert into USERS2 values (?, ?)")` but since Slick 3 I find the plain SQL API very hard to use (as in crippled/lacking, or if not, at least the online doc is lacking detail ?). This is an example of something I don't know how to do with new API. I'd appreciate your advice. Thank you.